Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java

```diff
@@ -617,13 +621,20 @@ public static void updateTierAndAC(SheetAdapter.SheetViewHolder holder, SheetMod
     @SuppressLint("SetTextI18n")
     public static void updateRateView(SheetAdapter.SheetViewHolder holder, SheetModel model) {
         double baseFare = model.getBaseFare() + model.getOfferedPrice() - model.getTollCharges();
-        float dist = model.getDistanceToBeCovFloat()/1000;
-        String rate = RideRequestUtils.getUptoDecStr((float) (baseFare/dist), 1);
+        float dist = model.getDistanceToBeCovFloat() / 1000;
+        String rate;
+
+        if (dist == 0.0f) {
+            rate = "NA";
+        } else {
+            rate = RideRequestUtils.getUptoDecStr((float) (baseFare / dist), 1);
+        }
+
         String currency = model.getCurrency();
-        String rateVal = dist > 0 ? (currency + rate) + "/km" : "NA";
-        holder.rateText.setText("Rate: " + rateVal);
+        holder.rateText.setText("Rate: " + currency + rate + "/km");
```

Comment:
@slow-codex if the rate is "NA" then I think it is better to not display this field, instead of showing "NA".

Reply:
Sure, we can set the visibility to gone in case of `"NA"`, but this will affect other merchants as well (because previously also we used to show `"NA"`). Can we do it in a seperate PR because this one is solely to sync the release branch of YS to main.